### PR TITLE
delete obsolete data in db

### PR DESCRIPTION
**Pull request by hougangliu**
Fixes: #311

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/315)

### CODE REVIEW

**Comment by YujiOshima** [comment]
Thanks! A comment. When we delete a study, related resources( trials, workers ) should be also deleted.

**Comment by hougangliu** [comment]
> Thanks! A comment. When we delete a study, related resources( trials, workers ) should be also deleted.

yes, all other resources will be deleted in db owning  to `FOREIGN KEY(study_id) REFERENCES studies(id) ON DELETE CASCADE`-like

**Comment by YujiOshima** [comment]
Good. Then please add a test for delete study to https://github.com/kubeflow/katib/blob/master/test/e2e/test-client.go

**Comment by hougangliu** [comment]
> Good. Then please add a test for delete study to https://github.com/kubeflow/katib/blob/master/test/e2e/test-client.go

OK thanks

File: test/e2e/test-client.go [hunk 1]
```diff
@@ -151,6 +152,17 @@ func CreateStudy(c api.ManagerClient) string {
...
 
+func DeleteStudy(c api.ManagerClient, studyID string) {
+	ctx := context.Background()
+	deleteStudyreq := &api.DeleteStudyRequest{
+		StudyId: studyID,
+	}
+	if _, err := c.DeleteStudy(ctx, deleteStudyreq); err != nil {
+		log.Fatalf("DeleteStudy error %v", err)
+	}
+	log.Printf("Study %s is deleted", studyID)
```
Comment (YujiOshima) [inline test/e2e/test-client.go]: You should check resources (study, trial, worker) are deleted by GetStudy or GetWorkers.

Comment (hougangliu) [reply test/e2e/test-client.go]: Thanks! Done

**Comment by YujiOshima** [comment]
@hougangliu Thank you!
/lgtm
/approve

**Pull request merged**